sdks/java/extensions/ml/src/main/java/org/apache/beam/sdk/extensions/ml/CloudVision.java

```diff
@@ -0,0 +1,238 @@
+
+/**
+ * Factory class for implementations of {@link AnnotateImages}.
+ *
+ * <p>Example usage:
+ *
+ * <pre>
+ * pipeline
+ *  .apply(Create.of(IMAGE_URI))
+ *  .apply(CloudVision.annotateImagesFromGcsUri(sideInputWithContext,
+ *         features, 1));
+ * </pre>
+ */
+public class CloudVision {
+
+  /**
+   * Creates a {@link org.apache.beam.sdk.transforms.PTransform} that annotates images from their
+   * GCS addresses.
+   *
+   * @param contextSideInput optional side input with contexts for select images.
+   * @param features annotation features that should be passed to the API
+   * @param batchSize request batch size to be sent to API. Max 5.
+   * @return the PTransform.
+   */
+  public static AnnotateImagesFromGcsUri annotateImagesFromGcsUri(
+      PCollectionView<Map<String, ImageContext>> contextSideInput,
+      List<Feature> features,
+      long batchSize) {
+    return new AnnotateImagesFromGcsUri(contextSideInput, features, batchSize);
+  }
+
+  /**
+   * Creates a {@link org.apache.beam.sdk.transforms.PTransform} that annotates images from their
+   * contents encoded in {@link ByteString}s.
+   *
+   * @param contextSideInput optional side input with contexts for select images.
+   * @param features annotation features that should be passed to the API
+   * @param batchSize request batch size to be sent to API. Max 5.
+   * @return the PTransform.
+   */
+  public static AnnotateImagesFromBytes annotateImagesFromBytes(
+      PCollectionView<Map<ByteString, ImageContext>> contextSideInput,
+      List<Feature> features,
+      long batchSize) {
+    return new AnnotateImagesFromBytes(contextSideInput, features, batchSize);
+  }
+
+  /**
+   * Creates a {@link org.apache.beam.sdk.transforms.PTransform} that annotates images from KVs of
+   * their GCS addresses in Strings and {@link ImageContext} for each image.
+   *
+   * @param features annotation features that should be passed to the API
+   * @param batchSize request batch size to be sent to API. Max 5.
+   * @return the PTransform.
+   */
+  public static AnnotateImagesFromBytesWithContext annotateImagesFromBytesWithContext(
+      List<Feature> features, long batchSize) {
+    return new AnnotateImagesFromBytesWithContext(features, batchSize);
+  }
+
+  /**
+   * Creates a {@link org.apache.beam.sdk.transforms.PTransform} that annotates images from KVs of
+   * their String-encoded contents and {@link ImageContext} for each image.
+   *
+   * @param features annotation features that should be passed to the API
+   * @param batchSize request batch size to be sent to API. Max 5.
+   * @return the PTransform.
+   */
+  public static AnnotateImagesFromGcsUriWithContext annotateImagesFromGcsUriWithContext(
+      List<Feature> features, long batchSize) {
+    return new AnnotateImagesFromGcsUriWithContext(features, batchSize);
+  }
+
+  /**
+   * Implementation of {@link AnnotateImages} that accepts {@link String} (image URI on GCS) with
```

Comment:
I think you can drop 'Implementaiton of' since we get this information from the class signature. It would be helpful to know what this implementation does differently than other implementations. The name of the class is very descriptive which is great, you could even start the comment with it to the effect of 'Annotates images from a GCS URI. (then a bit more detail)'.

Reply:
Good idea, thanks